### ISSUE TITLE
Object literal may only specify known properties, and 'origin' does not exist in type 'Partial<ModuleOptions>'

### ISSUE THREAD

**Issue by Timothy16** [nuxt/nuxt#21839]
### Environment

Operating System: Windows_NT
- Node Version:     v18.16.1
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.1
- Package Manager:  npm@9.6.7
- Builder:          vite
- User Config:      devtools, runtimeConfig, app, css, build, modules, auth, vite
- Runtime Modules:  @sidebase/nuxt-auth@0.6.0-beta.3, @pinia/nuxt@0.4.11
- Build Modules:    -

### Reproduction

Type '{ origin: string; basePath: string; enableGlobalAppMiddleware: true; }' is not assignable to type 'Partial<ModuleOptions>'.
  Object literal may only specify known properties, and 'origin' does not exist in type 'Partial<ModuleOptions>'.ts(2322)
schema.d.ts(4, 5): The expected type comes from property 'auth' which is declared here on type 'InputConfig<NuxtConfig, ConfigLayerMeta>'

### Describe the bug

The auth origin is heightening an error that is causing the project not to build in production 

<img width="616" alt="error" src="https://github.com/nuxt/nuxt/assets/55112043/2089fff8-b07a-400a-ac05-07a4968d2c9c">

**Comment by danielroe** [comment]
Note: this seems like an issue with the `@sidebase/nuxt-auth` module rather than Nuxt.

**Comment by danielroe** [comment]
I think you either need to provide a reproduction here or raise an issue on the nuxt-auth repo (and they would probably also apprciate a reproduction).

**Comment by Firstrings** [comment]
> 
Thank you for your reply. This project is a brand new one created from Nuxi, and I have not made any modifications or added custom content to it. I am also curious about why there is such a mistake.

Afterwards, I fixed this issue by downloading the @ nuxt/bridge package and compared the ts description of defineNuxtConfig.

Here is my guess:

Firstly, this error is only a syntax prompt error in TS. If after executing NPM run dev, neither Nuxt nor Vue reports an error at runtime, it proves to some extent that this writing is supported.
Secondly, I compared the defineNuxtConfig function in the initial and bridge packages. It was found that the relevant config attribute was explicitly declared in the bridge, But there is no similar description of ts in the initial code generated by nuxti.
Therefore, I believe it may be that the ts5.0.4 version does not support non explicit specifying, but it is currently uncertain
